data/enemy/enemies.py

```diff
@@ -0,0 +1,193 @@
+# ruff: noqa
+from typing import Self
+from data.enemy.enemies import *
+
+
+class Enemies:
+    MAPPINGS = {
+        "f4032b2323bc31d4590cf5197db3c3f1": Abstarak(),
```

Comment:
i'm assuming this may instantiate every enemy when we call Enemies().get(guid)

Is it possible in python to reference the object but not init it? then init it on the get?

We will need a new enemy object on every request - the memory code will check unique ids every frame, and if one doesn't exist it will use the guid to new an object -
it will also instead remove objects where the uid no longer exists.

Reply:
I'm a little uncertain what you are trying to do here.

What the code will do, is that on boot of the program (before get is every called), the static dict MAPPINGS will be initialized (once). This means all the init functions will be called and the objects populated with data.

There is a single instance of this data, so if you for some reason change the data on the object (not recommended), it will change for the next time you call get as well.

The correct way to call this is not `Enemies().get(guid)` (though there's no actual data to initialize there, so I guess that will work too), but `Enemies.MAPPINGS.get(guid)`

Reply:
So you're saying if i use the dict get it wont copy either and it will just be a reference?

In that case is there a solution for getting a fresh object each time?

Reply:
by default, you will get by reference. If you want one object from the dict as a new object, you can use this syntax:

```
reference = Enemies.MAPPINGS.get(guid)
new_object = reference.copy()
```
`new_object` is a shallow copy (so fine in this case, but it would copy any complex objects as references) and can be modified without changing the entry in the dict.